app/src/test/java/org/simple/clinic/home/overdue/phonemask/PhoneMaskBottomSheetControllerTest.kt

```diff
@@ -207,42 +153,32 @@ class PhoneMaskBottomSheetControllerTest {
 
   @Test
   fun `when the screen is created, the patient details must be populated`() {
-    // given
-    val patientSubject = PublishSubject.create<Optional<Patient>>()
-    val patientPhoneNumberSubject = PublishSubject.create<Optional<PatientPhoneNumber>>()
-    whenever(patientRepository.patient(patientUuid)).doReturn(patientSubject)
-    whenever(patientRepository.phoneNumber(patientUuid)).doReturn(patientPhoneNumberSubject)
-
-    val patientName = "Anish Acharya"
-    val gender = Male
-    val number = "1234567890"
-
-    val patient = PatientMocker.patient(
-        uuid = patientUuid,
-        fullName = patientName,
-        gender = gender,
-        dateOfBirth = LocalDate.of(1940, Month.JANUARY, 1),
-        age = null
-    )
-    val phoneNumber = PatientMocker.phoneNumber(
-        uuid = UUID.fromString("fe364a55-118a-4f0f-bb64-b7b0ca2e8282"),
-        patientUuid = patientUuid,
-        number = number
-    )
-
     // when
-    uiEvents.onNext(PhoneMaskBottomSheetCreated(patientUuid))
-    patientSubject.onNext(patient.toOptional())
-    patientPhoneNumberSubject.onNext(phoneNumber.toOptional())
+    sheetCreated()
 
     // then
     val expectedPatientDetails = PatientDetails(
-        phoneNumber = number,
-        name = patientName,
-        gender = gender,
+        phoneNumber = phoneNumber.number,
+        name = patient.fullName,
+        gender = patient.gender,
         // Test UTC clock is set to EPOCH
         age = 30
     )
     verify(screen).setupView(expectedPatientDetails)
   }
+
+  private fun sheetCreated() {
+    whenever(patientRepository.patient(patientUuid)).doReturn(Observable.just(patient.toOptional()))
+    whenever(patientRepository.phoneNumber(patientUuid)).doReturn(Observable.just(phoneNumber.toOptional()))
```

Comment:
Instead of implicitly using the global `patient` and `phoneNumber`, can we pass them to this method?
That will give the callers more control and the data being used for a test case will be more explicit.
We can still pass the global values.
Wdyt?

Reply:
Makes sense. I will use the global properties as default parameters so that the change is minimal.

Reply:
Done.